slsa/provenance.go

```diff
@@ -0,0 +1,123 @@
+package slsa
+
+import (
+	"runtime"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	slsa "github.com/in-toto/in-toto-golang/in_toto/slsa_provenance/v0.2"
+
+	"github.com/slsa-framework/slsa-github-generator/github"
+)
+
+// GithubHostedActionsBuilderID is the builder ID for Github hosted actions.
+const GithubHostedActionsBuilderID = "https://github.com/Attestations/GitHubHostedActions@v1"
+
+// WorkflowRun contains information about the build run including the builder,
+// build invocation, and parameters.
+type WorkflowRun struct {
+	// Subjects is information about the generated artifacts.
+	Subjects []intoto.Subject
+
+	// BuildType indicates the type of build that was done. More importantly it
+	// also specifies the format of the BuildConfig.
+	BuildType string
+
+	// BuildConfig is metadata about the build.
+	BuildConfig interface{}
```

Comment:
nit:  Since this is always nil, why not omit this?

And if you do that, then you can hardcode BuildType within provenance.go, which further simplifies the interface.

Reply:
I think this is a placeholder and the caller (ecosystem reusable workflow) will pass data of their choosing here. This way we'll be able to remove the provenance generator across ecosystem generators.

Reply:
The intention is also to allow folks to use these packages as a simple library for ecosystem/language-specific builders so allowing them to pass their own BuildConfig is desirable. BuildType needs to be configurable for the same reason.